beluga/include/beluga/views/take_while_kld.hpp

```diff
@@ -95,58 +97,103 @@ namespace detail {
 struct take_while_kld_fn {
   /// Overload that implements the take_while_kld algorithm.
   /**
-   * \tparam States An [input range](https://en.cppreference.com/w/cpp/ranges/input_range) with particle states.
+   * \tparam Range An [input range](https://en.cppreference.com/w/cpp/ranges/input_range) with particle states.
    * \tparam Hashes An [input range](https://en.cppreference.com/w/cpp/ranges/input_range) with particle hashes.
-   * \param states Source range from where to take elements.
+   * \param range Source range from where to take elements.
    * \param hashes Source range containing hashes (or bucket ids) to compute the KLD condition.
    * \param min Minimum samples to take.
    * \param max Maximum samples to take.
    * \param epsilon See beluga::kld_condition() for details.
    * \param z See beluga::kld_condition() for details.
    */
-  template <class States, class Hashes, std::enable_if_t<ranges::input_range<Hashes>, int> = 0>
+  template <
+      class Range,
+      class Hashes,
+      std::enable_if_t<ranges::range<Range>, int> = 0,
+      std::enable_if_t<ranges::range<Hashes>, int> = 0>
   constexpr auto operator()(
-      States&& states,
+      Range&& range,
       Hashes&& hashes,
       std::size_t min,
       std::size_t max,
       double epsilon,
       double z = beluga::detail::kDefaultKldZ) const {
-    static_assert(ranges::input_range<States>);
+    static_assert(ranges::input_range<Range>);
     static_assert(ranges::input_range<Hashes>);
     static_assert(std::is_convertible_v<ranges::range_value_t<Hashes>, std::size_t>);
     const auto hash = [](const auto& p) { return std::get<1>(p); };
-    return ranges::views::zip(ranges::views::all(states), ranges::views::all(hashes)) |  //
-           ranges::views::take_while(beluga::kld_condition(min, epsilon, z), hash) |     //
-           ranges::views::take(max) |                                                    //
+    return ranges::views::zip(ranges::views::all(range), ranges::views::all(hashes)) |  //
+           beluga::views::forward |                                                     //
```

Comment:
@nahueespinosa IIUC we do this to constrain `take_while_kld` output to be a forward range (as `take_while` seems to mimic the given range traits). Is that true? If so, why? Order is arbitrary and orthogonal to KLD conditions.

Reply:
Yes, that's true. Hmm, now that you say it, I think you're right. I'll remove the forward_view.